app/javascript/src/components/Dashboard/Contacts/Table.jsx

```diff
@@ -1,23 +1,53 @@
 import React, { useState } from "react";
 
-import { Table as FormikTable } from "neetoui";
+import { Table as NeetoUITable, Toastr } from "neetoui";
+import { buildRowData, updateContactEntity } from "utils";
 
-import { buildRowData } from "utils/index";
+import { COLUMN_DATA } from "./constants";
+import Edit from "./Pane/Edit";
 
-import { TABLE_COLUMN_DATA } from "./constants";
+const Table = ({
+  contacts = [],
+  setContacts,
+  selectedContactIds,
+  setSelectedContactIds,
+}) => {
+  const [pageNumber, setPageNumber] = useState(1);
+  const [showEditPane, setShowEditPane] = useState(false);
+  const [selectedContact, setSelectedContact] = useState({});
 
-const Table = ({ contacts = [] }) => {
-  const [page, setPage] = useState(1);
-  const TABLE_ROW_DATA = buildRowData(contacts);
+  const updateContact = (id, values) => {
+    setContacts(contacts =>
+      contacts.map(contact =>
+        contact.id === id ? updateContactEntity({ id, values }) : contact
+      )
+    );
```

Comment:
https://www.loom.com/share/6dc2e3019cfb4fb091374ddc77e36426

Reply:
I will move those functions to the closest scopes. But regarding the `getNewKey`, the following [comment](https://github.com/vivek-viswam-rv/neeto-ui-challenge/pull/20#discussion_r983528378) answers why.

Reply:
Yedin, please check out this video 
https://www.loom.com/share/9a1a67dcbfb44a9199347512219d65b4

The above video shows the latest commit. `INITIAL_VALUES` has been renamed to `EMPTY_CONTACT`.
Please let me know if anything needs to be done.🙂